exe/Main.hs

```diff
@@ -48,7 +48,7 @@ replaceCommandWithExit :: [String] -> [String]
 replaceCommandWithExit = (<> ["--command", "sh", "-c", "exit"]) . takeWhile (\x -> x /= "--command" && x /= "-c")
 
 withJSON :: [String] -> [String]
-withJSON x = "-v" : "--log-format" : "internal-json" : x
+withJSON x = x <> ["-v", "--log-format", "internal-json"]
```

Comment:
What is the purpose of this change?

From my understanding this means, that the user can’t override the debug-level anymore.

Reply:
The purpose is to make the NIX_GET_COMPLETIONS number correct. It is a number indicating which word of the input is being completed (eg: tabbing on a word in the middle of a command). Thus means we can use the Nix completion logic as-is.

Another approach could be to adjust that logic in the completion code with a +3 to account for the extra args.

Reply:
The completion code has no clue how we will be calling Nix, right? I can’t imagine a completion that would behave differently because of those two extra arguments. I vote for not changing the Haskell, but leaving the completion like it is. That seems good enough.

The bigger problem is, that this will complete e.g. `nom copy` although nom does not support that.

Reply:
The way the Nix completions works is that we pass in the argument number of the thing being completed.  So for example, you can run this directly to experiment...

```
[tom@tframe:~]$ NIX_GET_COMPLETIONS=2 nix build nixpkgs#hello
attrs
nixpkgs#hello
nixpkgs#hello-unfree
nixpkgs#hello-wayland
```
and it works if you have the cursor in the middle of the line; notice the "2".
```
[tom@tframe:~]$ NIX_GET_COMPLETIONS=2 nix build nixpkgs#hello --json -L
attrs
nixpkgs#hello
nixpkgs#hello-unfree
nixpkgs#hello-wayland
```

Because nom injects three extra arguments, the end result would be:
```
[tom@tframe:~]$ NIX_GET_COMPLETIONS=2 nix build -v --log-format internal-json nixpkgs#hello<TAB>
```

Which would complete the "-v" argument, not the expected one. The bash completer logic in _complete_nix does this. The alternative I mentioned (and was my original implementation) is to copy the nix completion logic, make a specific one for nom and modify it to adjust for the extra arguments:
https://github.com/NixOS/nix/blob/f0180487a0e4c0091b46cb1469c44144f5400240/misc/bash/completion.sh#L18

```diff
- done < <(NIX_GET_COMPLETIONS=$cword "${words[@]}" 2>/dev/null)
+ done < <(NIX_GET_COMPLETIONS=$(( cword + 3 )) "${words[@]}" 2>/dev/null)
```

Lastly, you can write your own completion handler, and call nix correctly yourself and write a completions/completion.bash to use that. This gives you the most control, but less re-use.

Reply:
Yeah, but what I am saying is that the completion engine has no clue that we will be calling `nix build` with `-v --log-format internal-json`. When you type `nom build nixpkgs#hel<TAB>` it will use the completion for `nix build nixpkgs#hel` and then after the user selected nixpkgs#hello and pressed enter we can insert flags wherever we want. That can’t mess with the completion.

Reply:
Ah! Good point.Then I don't know why I I had to add this. Perhaps left over from the previous style of implementation. I'll remove.

Reply:
No, now i need to back on that statement. The original exec `nom` is called by _complete_nix bash function in `"${words[@]}`, so we do need a correction somewhere. I updated it to another approach where `nom` detects this case can calls the underlying nix without the extra args.